index.js

```diff
@@ -2,25 +2,36 @@
 const { ProvidePlugin } = require("webpack")
 const filterObject = require("filter-obj")
 
-const excludeObjectKeys = (object, excludeKeys) => filterObject(object, key => !excludeKeys.includes(key))
+const filterObjectKeys = (includeAliases, excludeAliases) => {
+	const isInclude = includeAliases.length > 0
+	const keys = isInclude ? includeAliases : excludeAliases
+	return object => filterObject(object, key => keys.includes(key) === isInclude)
+}
 
 module.exports = class NodePolyfillPlugin {
 	constructor(options = {}) {
 		this.options = {
 			excludeAliases: [],
+			includeAliases: [],
 			...options
 		}
+
+		if (this.options.includeAliases.length > 0 && this.options.excludeAliases.length > 0) {
+			throw new Error("excludeAliases and includeAliases are mutually exclusive!")
+		}
 	}
 
 	apply(compiler) {
-		compiler.options.plugins.push(new ProvidePlugin(excludeObjectKeys({
+		const filterFunc = filterObjectKeys(this.options.includeAliases, this.options.excludeAliases)
```

Comment:
```suggestion
		const filter = filterObjectKeys(this.options.includeAliases, this.options.excludeAliases)
```

Requires other changes